checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakAnnotatedTypeFactory.java

```diff
@@ -231,7 +231,9 @@ protected ResourceLeakAnalysis createFlowAnalysis() {
    * @param tree the tree of the expression the tempvar represents
    */
   /* package-private */ void addTempVar(LocalVariableNode tmpVar, Tree tree) {
-    tempVarToTree.put(tmpVar, tree);
+    if (!tempVarToTree.containsValue(tree)) {
```

Comment:
I'm a little concerned about this change.  If `addTempVar()` is being called, the caller has already created a new `tmpVar` for the `tree`.  It seems problematic if `tmpVar` gets used elsewhere but doesn't get added to the `tempVarToTree` mapping.  What do you think of this concern?

Reply:
I don't think it makes any problem because each node has to have a unique tmpVar which is defined in MustCallAnnotatedTypeFactory. But the problem here is that in ResourceLeakTransfer in visitTernaryExpression we don't check if there exist a tmpVar for the node like [this](https://github.com/Nargeshdb/checker-framework/blob/fix-5136/checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakTransfer.java#L140).

Reply:
Ok, assuming the tests pass, I am ok with this change for now.  I'll finish reviewing once CI passes (I think there is a formatting error)